sdk/core/core-rest-pipeline/src/fetchHttpClient.ts

```diff
@@ -0,0 +1,237 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+/// <reference lib="dom" />
+
+import { AbortError } from "@azure/abort-controller";
+import {
+  HttpClient,
+  HttpHeaders as PipelineHeaders,
+  PipelineRequest,
+  PipelineResponse,
+  TransferProgressEvent,
+} from "./interfaces";
+import { RestError } from "./restError";
+import { createHttpHeaders } from "./httpHeaders";
+
+/**
+ * Checks if the body is a NodeReadable stream which is not supported in Browsers
+ */
+function isNodeReadableStream(body: any): body is NodeJS.ReadableStream {
+  return body && typeof body.pipe === "function";
+}
+
+/**
+ * Checks if the body is a ReadableStream supported by browsers
+ */
+function isReadableStream(body: unknown): body is ReadableStream {
+  return Boolean(
+    body &&
+      typeof (body as ReadableStream).getReader === "function" &&
+      typeof (body as ReadableStream).tee === "function"
+  );
+}
+
+/**
+ * A HttpClient implementation that uses window.fetch to send HTTP requests.
+ * @internal
+ */
+class FetchHttpClient implements HttpClient {
+  /**
+   * Makes a request over an underlying transport layer and returns the response.
+   * @param request - The request to be made.
+   */
+  public async sendRequest(request: PipelineRequest): Promise<PipelineResponse> {
+    const url = new URL(request.url);
+    const isInsecure = url.protocol !== "https:";
+
+    if (isInsecure && !request.allowInsecureConnection) {
+      throw new Error(`Cannot connect to ${request.url} while allowInsecureConnection is false.`);
+    }
+
+    if (request.proxySettings) {
+      throw new Error("HTTP proxy is not supported in browser environment");
+    }
+
+    try {
+      return await makeRequest(request);
+    } catch (e) {
+      throw getError(e, request);
+    }
+  }
+}
+
+/**
+ * Sends a request
+ */
+async function makeRequest(request: PipelineRequest): Promise<PipelineResponse> {
+  const abortController = handleAbortSignal(request);
+  const headers = buildFetchHeaders(request.headers);
+
+  const requestBody = buildRequestBody(request);
+
+  const response = await fetch(request.url, {
+    body: requestBody,
+    method: request.method,
+    headers: headers,
+    signal: abortController.signal,
+    credentials: request.withCredentials ? "include" : "same-origin",
```

Comment:
I think we need a small doc update for `withCredentials` since we do send them on same origin and the current description doesn't make that obvious

Reply:
Updating to `Indicates whether the user agent should send cookies from the other domain in the case of cross-origin requests`